src/Unused/GitContext.hs

```diff
@@ -29,5 +29,6 @@ logToGitContext =
 
 gitLogSearchFor :: Int -> [String] -> IO GitOutput
 gitLogSearchFor commitCount ts = do
-  (_, results, _) <- P.readProcessWithExitCode "git" ["log", "-G", L.intercalate "|" ts, "--oneline", "-n", show commitCount] ""
+  let gitLogFlag = if length ts == 1 then "-S" else "-G"
+  (_, results, _) <- P.readProcessWithExitCode "git" ["log", gitLogFlag, L.intercalate "|" ts, "--oneline", "-n", show commitCount] ""
```

Comment:
I wonder if it might be easier to read if we pattern matched on `ts` here instead of using a conditional? That would also make the `L.intercalate` call a bit more explicit and might make it clearer why we need `-G` in the first place. Possibly something along the lines of:

```haskell
gitLogSearchFor :: Int -> [String] -> IO GitOutput
gitLogSearchFor commitCount ts = do
    (_, results, _) <- P.readProcessWithExitCode "git" (gitCommand commitCount ts) ""
    return $ GitOutput results
  where
    gitCommand :: Int [String] -> [String]
    gitCommand commitCount [t] = ["log", "-S", t, "--oneline", "-n", show commitCount]
    gitCommand commitCount ts = ["log", "-G", L.intercalate "|" ts, "--oneline", "-n", show commitCount]
```

Reply:
I like the idea of pattern-matching on the array for this!